test.js

```diff
@@ -108,3 +109,26 @@ test('execa() returns a promise with kill() and pid', t => {
 	t.is(typeof promise.kill, 'function');
 	t.is(typeof promise.pid, 'number');
 });
+
+test('observable on stdin', t => {
+	t.plan(2);
+
+	return m('stdin', [], {stdout: 'observable', input: 'hello\ngoodbye\nhello'})
+		.stdout
+		.filter(line => /h/.test(line))
+		.map(line => line.toUpperCase())
+		.forEach(line => t.is(line, 'HELLO'));
```

Comment:
Overdone example here, just showing the power of `Observables`.

Understand that the `.filter`, `.map` and `.forEach` methods are not synchronous like Array methods. And they fire as lines are encountered. The second call to the `.filter` method could happen long after the first call to `.forEach`.